plugins/out_stackdriver/stackdriver.c

```diff
@@ -2834,57 +2841,61 @@ static void cb_stackdriver_flush(struct flb_event_chunk *event_chunk,
                                                ts,
                                                (int) event_chunk->total_events,
                                                grpc_status_counts);
 
-#ifdef FLB_HAVE_METRICS
             int failed_records = 0;
             if (ret_partial_success == 0) {
-                for (code = 0; code < GRPC_STATUS_CODES_SIZE; code++) {
-                    failed_records += grpc_status_counts[code];
-                    add_record_metrics(ctx, ts, grpc_status_counts[code], code);
-                }
-                cmt_counter_add(ctx->ins->cmt_dropped_records, ts,
-                                failed_records, 1, (char* []) {name});
-                int successful_records =
-                    (int) event_chunk->total_events - failed_records;
-                if (successful_records != 0) {
-                    add_record_metrics(ctx, ts, successful_records, 200);
+              for (code = 0; code < GRPC_STATUS_CODES_SIZE; code++) {
+                if (grpc_status_counts[code] != 0) {
+                  failed_records += grpc_status_counts[code];
                 }
-            } else {
-                add_record_metrics(ctx, ts, (int) event_chunk->total_events,
-                                   c->resp.status);
-                cmt_counter_add(ctx->ins->cmt_dropped_records, ts,
-                                (int) event_chunk->total_events, 1,
-                                (char* []) {name});
+              }
+              cmt_counter_add(ctx->ins->cmt_dropped_records, ts,
+                              failed_records, 1, (char* []) {name});
+              int successful_records =
+                  (int) event_chunk->total_events - failed_records;
+              if (successful_records != 0) {
+                add_record_metrics(ctx, ts, successful_records, 200, 0);
+              }
+            }
+            else {
+              add_record_metrics(ctx, ts, (int) event_chunk->total_events,
+                                 c->resp.status, -1);
+              cmt_counter_add(ctx->ins->cmt_dropped_records, ts,
+                              (int) event_chunk->total_events, 1,
+                              (char* []) {name});
             }
 #endif
-
-            flb_plg_warn(ctx->ins, "error\n%s", c->resp.payload);
-        }
-        else {
+          if (c->resp.status >= 400 && c->resp.status < 500) {
+            ret_code = FLB_ERROR;
+            flb_plg_warn(ctx->ins, "error: %s", c->resp.payload);
+          }
+          else {
             if (c->resp.payload_size > 0) {
-                /* we got an error */
-                flb_plg_warn(ctx->ins, "error\n%s", c->resp.payload);
+              /* we got an error */
+              flb_plg_warn(ctx->ins, "error: %s", c->resp.payload);
             }
             else {
-                flb_plg_debug(ctx->ins, "response\n%s", c->resp.payload);
+              flb_plg_debug(ctx->ins, "response: %s", c->resp.payload);
```

Comment:
Would this not always print an empty string, because `c->resp.payload_size`==0 in this branch?

Reply:
This is prior functionality.